esmvaltool/cmorizers/data/formatters/datasets/esrl.ncl

```diff
@@ -0,0 +1,208 @@
+  st_alt  = toint(str_get_field(stat(:), 5, ","))
+  nstats  = dimsizes(st_code)
+  delete(stat)
+
+  ; Loop over stations
+  do stID = 0, nstats - 1
+
+    infile = input_dir_path + "co2_" + str_lower(st_code(stID)) + \
+             "_surface-insitu_1_ccgg_month.txt"
+
+    if (.not.fileexists(infile)) then
+      infile = input_dir_path + "co2_" + str_lower(st_code(stID)) + \
+               "_surface-insitu_1_ccgg_MonthlyData.txt"
+    end if
+
+    if (fileexists(infile)) then
+
+      log_info("Processing " + VAR + " (" + st_code(stID) + ")")
+
+      raw = asciiread(infile, -1, "string")
+
+      head = str_get_field(raw(0), 3, " ")
+      if (typeof(head).eq. "string") then
+        head = str_get_field(raw(0), 4, " ")
+      end if
+
+      data = raw(toint(head):)
+      delete([/head, raw/])
+
+      ; Read site name
+      tmp = str_get_field(data(:), 1, " ")
+      if (.not.all(tmp(0).eq.tmp)) then
+          log_info("Inconsistent site name in " + infiles(ii))
+      else
+        site = tmp(0)
+      end if
+      delete(tmp)
+
+      ; Read time and extract full years only
+      year = str_squeeze(str_get_field(data(:), 2, " "))
+      month = str_squeeze(str_get_field(data(:), 3, " "))
+      month = sprinti("%2.2i", toint(month))
+      date = toint(year + month)
+      var = tofloat(str_get_field(data(:), 9, " "))
+      delete(data)
+
+      ; Check time range
+      if (min(toint(year)).lt.YEAR1 .or. max(toint(year)).gt.YEAR2) then
+        error_msg("f", DIAG_SCRIPT, "", "given time range (" + \
+                  YEAR1 + "-" + YEAR2 + ")" + \
+                  " not consistent with data (" + \
+                  min(toint(year)) + "-" + max(toint(year)) + ")")
+      end if
+      delete(year)
+      delete(month)
+
+      ; Create output variable
+      output = new((/dimsizes(timec), 1, 1/), float)
+      output!0 = "time"
+      output&time = timec
+      output!1 = "lon"
+      output&lon = st_lon(stID)
+      output!2 = "lat"
+      output&lat = st_lat(stID)
+      output@_FillValue = FILL
+
+      do ii = 0, dimsizes(var) - 1
+        output(ind(datec.eq.date(ii)), 0, 0) = (/var(ii)/)
+      end do
+      delete(var)
+      delete(date)
+      output = where(output.lt.0., output@_FillValue, output)
+
+      ; Format coordinates
+      format_coords(output, YEAR1 + "0101", YEAR2 + "1231", FREQ)
+
+      ; Set variable attributes (CMOR)
+      tmp = format_variable(output, VAR, CMOR_TABLE)
+      delete(output)
+      output = tmp
+      delete(tmp)
+
+      ; Calculate coordinate bounds
+      bounds = NewList("fifo")
+      time_bnds = guess_bounds_time(output&time, FREQ)
+      ListPush(bounds, time_bnds)
+
+      ; Set global attributes
+      gAtt = set_global_atts(OBSNAME, TIER, SOURCE, REF, COMMENT)
+
+      ; Outfile
+      VERSION = st_code(stID)
+      DATESTR = YEAR1 + "01-" + YEAR2 + "12"
+
+      fout = output_dir_path + str_join((/"OBS", OBSNAME, TYPE, VERSION, \
```

Comment:
cmorized filenames contain a blank space, i.e.: `OBS_ESRL_g round_BRW_Amon_co2s_197301-201912.nc`. There seems to be a minor issue in this script.

Reply:
Outdated and removed